taskcluster/app_services_taskgraph/job.py

```diff
@@ -0,0 +1,176 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+from __future__ import absolute_import, print_function, unicode_literals
+
+from taskgraph.transforms.job import run_job_using, configure_taskdesc_for_run
+from taskgraph.util import path
+from taskgraph.util.schema import Schema, taskref_or_string
+from voluptuous import Required, Optional
+from six import text_type
+
+from pipes import quote as shell_quote
+
+secret_schema = {
+    Required("name"): text_type,
+    Required("path"): text_type,
+    Required("key"): text_type,
+    Optional("json"): bool,
+}
+
+dummy_secret_schema = {
+    Required("content"): text_type,
+    Required("path"): text_type,
+    Optional("json"): bool,
+}
+
+gradlew_schema = Schema({
+    Required("using"): "gradlew",
+    Optional("pre-gradlew"): [[text_type]],
+    Required("gradlew"): [text_type],
+    Optional("post-gradlew"): [[text_type]],
+    # Base work directory used to set up the task.
+    Required("workdir"): text_type,
+    Optional("use-caches"): bool,
+    Optional("secrets"): [secret_schema],
+    Optional("dummy-secrets"): [dummy_secret_schema],
+})
+
+run_commands_schema = Schema({
+    Required("using"): "run-commands",
+    Optional("pre-commands"): [[text_type]],
+    Required("commands"): [[taskref_or_string]],
+    Required("workdir"): text_type,
+    Optional("use-caches"): bool,
+    Optional("secrets"): [secret_schema],
+    Optional("dummy-secrets"): [dummy_secret_schema],
+})
+
+
+@run_job_using("docker-worker", "run-commands", schema=run_commands_schema)
+def configure_run_commands_schema(config, job, taskdesc):
+    run = job["run"]
+    pre_commands = run.pop("pre-commands", [])
+    pre_commands += [
+        _generate_dummy_secret_command(secret) for secret in run.pop("dummy-secrets", [])
+    ]
+    pre_commands += [
+        _generate_secret_command(secret) for secret in run.get("secrets", [])
+    ]
+
+    all_commands = pre_commands + run.pop("commands", [])
+
+    run["command"] = _convert_commands_to_string(all_commands)
+    _inject_secrets_scopes(run, taskdesc)
+    _set_run_task_attributes(job)
+    configure_taskdesc_for_run(config, job, taskdesc, job["worker"]["implementation"])
+
+
+@run_job_using("docker-worker", "gradlew", schema=gradlew_schema)
+def configure_gradlew(config, job, taskdesc):
+    run = job["run"]
+    worker = taskdesc["worker"] = job["worker"]
+
+    # TODO: to uncomment later when we'll port over logic from bug 1622339
```

Comment:
Will address this later when I'll port over the logic from https://bugzilla.mozilla.org/show_bug.cgi?id=1622339